test/test-fs-event.c

```diff
@@ -824,44 +809,32 @@ TEST_IMPL(fs_event_close_in_callback) {
 
   loop = uv_default_loop();
 
+  fs_event_unlink_files(NULL);
   create_dir("watch_dir");
-  create_file("watch_dir/file1");
-  create_file("watch_dir/file2");
-  create_file("watch_dir/file3");
-  create_file("watch_dir/file4");
-  create_file("watch_dir/file5");
 
   r = uv_fs_event_init(loop, &fs_event);
   ASSERT(r == 0);
   r = uv_fs_event_start(&fs_event, fs_event_cb_close, "watch_dir", 0);
   ASSERT(r == 0);
 
-  /* Generate a couple of fs events. */
-  touch_file("watch_dir/file1");
-  touch_file("watch_dir/file2");
-  touch_file("watch_dir/file3");
-  touch_file("watch_dir/file4");
-  touch_file("watch_dir/file5");
+  r = uv_timer_init(loop, &timer);
+  ASSERT(r == 0);
+  r = uv_timer_start(&timer, fs_event_create_files, 100, 0);
```

Comment:
@bnoordhuis Good catch, thanks!

I've tried to reuse as much code as possible, so the timer is [now closed after the event loop is done](https://github.com/libuv/libuv/pull/1011/files).

Otherwise I would need to change `fs_event_create_files` to close the handle when there's no file creation left to do, but that would mean changing a lot more code to adapt to that.

I could also create a separate function to create the files needed for this specific test, and close the handle when all files have been created. But that would duplicate some of the code in `fs_event_create_files`.